src/fhirtypes/ElementDefinition.ts

```diff
@@ -417,7 +417,9 @@ export class ElementDefinition {
       .split('.')
       .map(p => {
         const i = p.indexOf('[x]:');
-        return i > -1 ? p.slice(i + 4) : p;
+        const isChoiceSlice =
+          i > -1 ? this.type?.some(type => p.slice(i + 4).endsWith(upperFirst(type.code))) : false;
+        return i > -1 && isChoiceSlice ? p.slice(i + 4) : p;
```

Comment:
Isn't just `isChoiceSlice` sufficient here, since when `isChoiceSlice` is `true`, `i > -1` is also always true, and when `isChoiceSlice` is `false`, it doesn't matter if `i >  -1`, since we will just return `p` either way.